src/nodes/Geometry3D/IndexedFaceSet.js

```diff
@@ -932,111 +943,18 @@ x3dom.registerNodeType(
                                             this._mesh._texCoords[0].push(texCoords[t2].z);
                                         }
                                     }
-
```

Comment:
Before the new indices (p0,p1,p2) where computed before the push of the values, so that why you need to separate the case : 
creation of the first triangle, t=2 -> update p2  -> push -> t=3
creation of the other triangle (t=3): update p0,p1,p2  -> push -> t=3 or t=-1

With the fix : I update p2, then push, then update p0 and p1
Now : update t = 2 -> p2  -> push -> update p0,p1 -> t=2 or t=-1